pkg/library/container/conversion.go

```diff
@@ -107,7 +110,18 @@ func devfileResourcesToContainerResources(devfileContainer *dw.ContainerComponen
 
 	cpuLimit := devfileContainer.CpuLimit
 	if cpuLimit == "" {
-		cpuLimit = constants.SidecarDefaultCpuLimit
+		// Set empty CPU limits when possible:
+		// 1. If there is no LimitRange in the namespace
+		// 2. CPU limits is not overridden
+		// See details at https://github.com/eclipse/che/issues/22198
+		limitRanges := &corev1.LimitRangeList{}
+		if err := k8sClient.List(context.TODO(), limitRanges, &client.ListOptions{Namespace: namespace}); err != nil {
+			return nil, err
+		} else if len(limitRanges.Items) == 0 {
+			cpuLimit = ""
+		} else {
+			cpuLimit = constants.SidecarDefaultCpuLimit
+		}
```

Comment:
The current behavior for DevWorkspace container components is to not set a limit -- `constants.SidecarDefaultCpuLimit` is `""`, which will lead to the conditional on line 126 not passing and so not setting a CPU limit.